### PR TITLE
autosharding content-topic config

### PR DESCRIPTION
**Pull request by chaitanyaprem**
# Description

New config to specify pubsub and contentTopics. 
ENR updation as per config. 

# Changes

- [x] Added new command flags for pubsub and contentTopics
- [x] Deprecated existing topic flag
- [x] Removed store-resume-peer flag and not invoking relevant code. Refer https://github.com/waku-org/nwaku/pull/1970
- [x] Added shard choice simulation test 
- [x] ENR update based on topics indicated

# Tests

- [x] Run serviceNode by passing specific content-topic "/0/toychat/2/huilong/proto"  and verified the ENR generated includes the appropriate rs value. 
`enr:-Jq4QAks1JpiJ8kjKXZbUvNEjDJb5F7u1xtIyb9XXZtVEhIvHhVpDb53CQpeknyWEr5dw0U-gVFqdm-4V-blSMvAomyGAYpKuzUMgmlkgnY0gmlwhMCoAQyCcnOFAAEBAAOJc2VjcDI1NmsxoQINUHz8nP1tgbfQv4-_Aqw-enrekJpmX6u0ASpuEUPnnoN0Y3CC6mCFd2FrdTIB`
- [x] Validate error is thrown in following scenarios
> -  If topics are passed that belong to different shard cluster
> - If named and sharded topics are passed together

### CODE REVIEW

**Comment by chaitanyaprem** [comment]
@richard-ramos , in nwaku if topic validation fails the node doesn't start with an error indicating named and static shards to be not used simultaneously. But in go-waku case we seem to be proceeding ahead and just logging a warning. 
Should we restrict this behaviour altogether or leave it as warning? 
If the node runs like this, anyways the ENR cannot be updated and hence is not useful actually as per the user expecations right. Or is this something specially done to handle status use-cases? Require your advice on how to proceed regarding this.

**Comment by richard-ramos** [comment]
> @richard-ramos , in nwaku if topic validation fails the node doesn't start with an error indicating named and static shards to be not used simultaneously. But in go-waku case we seem to be proceeding ahead and just logging a warning.
> Should we restrict this behaviour altogether or leave it as warning?

Oops! this was an oversight when I implemented that functionality. https://github.com/waku-org/go-waku/pull/699 fixes this.

**Review by richard-ramos — APPROVED** [review]

File: cmd/waku/node.go [hunk 1]
```diff
@@ -534,6 +537,37 @@ func loadPrivateKeyFromFile(path string, passwd string) (*ecdsa.PrivateKey, erro
 	return crypto.ToECDSA(pKey)
 }
 
+/*
```
Comment (richard-ramos) [inline cmd/waku/node.go:540]: What do you think if we instead remove this function completely and not only leave it commented. We can always rely on commit history to restore it back.

**Comment by chaitanyaprem** [comment]
> PR looks good to me. Will chat2 be updated in this PR as well or in a separate PR?

Ah...missed compiling and updating examples. Will do that as well in this PR only.

File: waku/v2/protocol/topic_test.go [hunk 2]
```diff
@@ -53,6 +55,63 @@ func TestContentTopicAndSharding(t *testing.T) {
...
+		counts[pubsub.Shard()] += 1
+	}
+
+	for i := 0; i < GenerationZeroShardsCount; i++ {
+		t.Logf("Counts at index %d is %d", i, counts[i])
+	}
+
+	// Then
+	for i := 1; i < GenerationZeroShardsCount; i++ {
+		if float64(counts[i-1]) <= (float64(counts[i])*1.05) &&
```
[the remark this answers is not in the document]
Comment (chaitanyaprem) [reply waku/v2/protocol/topic_test.go:106]: @SionoiS , These checks for shard distribution validation seem duplicate. 
Was your intention to validate something else?

[the remark this answers is not in the document]
Comment (chaitanyaprem) [reply waku/v2/protocol/topic_test.go:106]: @harsh-98 , merging this for now. Will resolve this comment based on further discussion.

**Pull request merged**